libraries/SD/src/SD.h

```diff
@@ -1,154 +1,134 @@
 /*
+ SD.h - A thin shim for Arduino ESP8266 Filesystems
+ Copyright (c) 2019 Earle F. Philhower, III.  All rights reserved.
 
- SD - a slightly more friendly wrapper for sdfatlib
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2.1 of the License, or (at your option) any later version.
 
- This library aims to expose a subset of SD card functionality
- in the form of a higher level "wrapper" object.
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
 
- License: GNU General Public License V3
-          (Because sdfatlib is licensed with this.)
-
- (C) Copyright 2010 SparkFun Electronics
-
- */
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
 
 #ifndef __SD_H__
 #define __SD_H__
 
 #include <Arduino.h>
+#include <FS.h>
+#include <SDFS.h>
 
-#include <utility/SdFat.h>
-#include <utility/SdFatUtil.h>
+#undef FILE_READ
+#define FILE_READ sdfat::O_READ
+#undef FILE_WRITE
+#define FILE_WRITE (sdfat::O_READ | sdfat::O_WRITE | sdfat::O_CREAT)
 
-#define FILE_READ O_READ
-#define FILE_WRITE (O_READ | O_WRITE | O_CREAT)
+class SDClass {
+public:
+    boolean begin(uint8_t csPin, SPISettings cfg = SPI_HALF_SPEED) {
+        auto fsCfg = SDFSConfig(csPin, cfg);
+	SDFS.setConfig(&fsCfg);
```

Comment:
If setConfig() were to receive by const ref, this wouldn't need to pass the address of fsCfg, which looks weird.

Reply:
Ditto, and now we can make a SDFSConfig object directly in-line with the setConfig method.